lib/PICA/Path.pm

```diff
@@ -261,6 +447,270 @@ without inspection field and occurrence values.
+    
+    # get two subfields by code
+    $path = PICA::Path->new('021Aad');
+    $match = $path->match($record);
+    is($match, 'TitleSupplement');
+    
+    $path = PICA::Path->new('021Ada');
+    $match = $path->match($record);
+    is($match, 'TitleSupplement');
+    
+    # get two subfields by code in specific order
+    $path = PICA::Path->new('021Ada');
+    $match = $path->match($record, {pluck => 1});
+    is($match, 'SupplementTitle');
+    
+    # join subfields
+    $path = PICA::Path->new('021Ada');
+    $match = $path->match($record, {pluck => 1, join => ' '});
+    is($match, 'Supplement Title');
+
+Option C<split> creates a list out of subfields:
+
+    # split subfields to list
+    $path = PICA::Path->new('021Ada');
+    $match = $path->match($record, {split => 1});
+    is_deeply($match, ['Title', 'Supplement']);
+
+Option C<nested_arrays> creates a list for every field found:
+
+    # split fields to lists
+    $path = PICA::Path->new('021Ada');
+    $match = $path->match($record, {split => 1, nested_arrays => 1});
+    is_deeply($match, [['Title', 'Supplement']]);
+
+=head2 Match single field with subfield repetition
+
+Field C<009Q> has repeated subfields:
+
+    $record = {
+        record => [
+            ...
+            [ '009Q', '', 'u', 'http://example.org/', 'x', 'A', 'z', 'B', 'z', 'C' ]
+            ...
+        ]
+    };
+    
+    # get all subfields
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record);
+    is($match, 'http://example.orgABC');
+    
+    # get repeated subfields
+    $path = PICA::Path->new('009Qz');
+    $match = $path->match($record);
+    is($match, 'BC');
+
+Option C<split> creates a list out of subfields:
+
+    # split subfields to list
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record, {split => 1});
+    is_deeply($match, ['http://example.org', 'A', 'B', 'C']);
+    
+    # split subfields to list
+    $path = PICA::Path->new('009Qz');
+    $match = $path->match($record, {split => 1});
+    is_deeply($match, ['B', 'C']);
+
+Option C<nested_arrays> creates a list for every field found:
+
+    # split fields to lists
+    $path = PICA::Path->new('009Qz');
+    $match = $path->match($record, {split => 1, nested_arrays => 1});
+    is_deeply($match, [['B', 'C']]);
+
+=head2 Match repeated Field with no subfield repetition
+
+Field C<005A> is repeated:
+
+    $record = {
+        record => [
+            [ '005A', '', '0', '1234-5678' ],
+            [ '005A', '', '0', '1011-1213' ],
+            ...
+        ]
+    };
+    
+    # get all subfields
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record);
+    is($match, '1234-56781011-1213');
+    
+    # get subfields by code
+    $path = PICA::Path->new('009Q');
+    $match = $path->match($record);
+    is($match, '1234-56781011-1213');
+
+Option C<split> creates a list out of subfields:
+
+    # split subfields to list
+    $path = PICA::Path->new('005A0');
```

Comment:
I prefer to include the optional `$` to better make clear that `0` is a subfield code: `005A$0`.